packages/drivers/routerlicious-driver/src/queryStringUtils.ts

```diff
@@ -0,0 +1,21 @@
+export type QueryStringType = Record<string, string | number | boolean>;
+
+/**
+ * Generates URL string from the given URL and query parameters.
+ * @param url - URL to which query params should be appended. Can include base/default query params.
+ * @param queryParams - Query parameters from which to create a query. Will override any query params in url.
+ */
+export function buildUrlWithQueryString(
```

Comment:
nit: Consider renaming to something like `addOrUpdateQueryParams` .. May make it more readable in references.